### ISSUE TITLE
Add support for mhlo.dynamic-update-slice

### ISSUE THREAD

**Issue by hanhanW** [google/iree#5982]

**Comment by GMNGeoffrey** [comment]
In https://github.com/google/iree/issues/5989 I'm running into a failure in lowering an `mhlo.dynamic-update-slice` through a stream. Is this the same issue you hit? It's getting lowered but results in IR unhandled by later passes.

**Comment by hanhanW** [comment]
I found the operation in the model and we don't have direct lowering support for `mhlo.dynamic-update-slice` to Linalg/std ops.

I looked into the codebas a bit, and found that it's coming from PrePartitioningConversionPass. The implementation is at

https://github.com/google/iree/blob/4ff7c453a20f6bce60d4441dca59aa6352a6f206/iree/compiler/Dialect/Flow/Conversion/HLOToFlow/ConvertHLOToFlow.cpp#L42-L59

It was added a long time ago, we should revisit it. Actually we miss some clamp ops in the lowering, but it seems not relevant. Eventually, we will have a pattern in MHLO repo instead and remove this one. We can have a similar lowering like `mhlo.dynamic-slice`, and use `subtensor_insert` instead of `subtensor`. But I feel we will have something more to plumb it through, and could run into the same failure you have or other failures.
